Library/Homebrew/extend/os/linux/extend/pathname.rb

```diff
@@ -0,0 +1,12 @@
+class Pathname
+  # @private
+  def elf?
+    read(4) == "\x7fELF"
+  end
+
+  # @private
+  def dynamic?
+    @which_file ||= which("file")
+    !@which_file.nil? && `#{@which_file} -b #{self}` =~ /dynamic|shared/
```

Comment:
I suggest…
```ruby
@which_file ||= which "file"
return false unless @which_file
popen_read(@which_file, "-b", to_path) =~ /dynamic|shared/
```

Reply:
I don't think we should return `false` because that would be incorrect.
We should, perhaps, `raise StandardError.new("file command is not available")` error in that case.
I'm also considering adding a `if/else`:
`if which("file")... elsif which("readelf") ... elseif which("ldd") ... else raise...`

Reply:
The current behaviour is to return `false` when `file` is not available. I understand your objection though. Trying first `readelf` and then `file` seems reasonable.

Reply:
> We should, perhaps, raise StandardError.new("file command is not available") error in that case.

Something like this sounds good or the `if`/`else`.

Reply:
If you're up for it, it wouldn't be too terribly difficult to parse the ELF header in Ruby code and determine whether the file is static or dynamic. The information isn't unfortunately at a fixed position, but it's not difficult to parse. Iterate through the vector of `Program header` entries. If you see a `PT_DYNAMIC` header, it's dynamic, otherwise it's static.

See https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#Program_header

If you're not into it, I don't have the time to implement a PR this week, but I could next week. I'd prefer to avoid having `brew` depend on external commands which may or may not be available.